transaction-service-api/src/main/java/pl/adamboguszewski/transactionservice/service/api/transaction/CreateTransactionRequest.java

```diff
@@ -0,0 +1,92 @@
+package pl.adamboguszewski.transactionservice.service.api.transaction;
+
+import lombok.Value;
+
+import javax.validation.constraints.NotNull;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+
+@Value
+public class CreateTransactionRequest {
+
+    @NotNull
+    UUID transactionId;
+
+    @NotNull
+    TransactionInfo transactionInfo;
+
+    @Value
+    public static class TransactionInfo {
+
+        @NotNull
+        String checkoutId;
+
+        @NotNull
+        LocalDateTime transactionTime;
+
+        @NotNull
+        List<TransactionPayment> transactionPayments;
```

Comment:
We have @NotEmpty annotation for collections too, don't use it here for now but remember for future development.